### PR TITLE
Create prow/autobump.sh and run every day

### PR DESCRIPTION
**Pull request by fejta**
/assign @BenTheElder @stevekuznetsov @ixdy @cjwagner 

* Move `push.sh` logic from `bump.sh` into its own file
  - Update `post-test-infra-push-prow` to use it
  - Make these work on mac
* Create `robots/pr-creator` to create/update a PR
  - `--match-title` will cause it to try and update an existing, open PR
* Create `prow/autobump.sh` to use `bump.sh --latest` and `pr-creator`
  - Always pushes to the user's `autobump` branch
  - Includes a link to the changes in the PR description

### CODE REVIEW

File: prow/autobump.sh [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+# * ensure there are no pending changes
+# * optionally activate GOOGLE_APPLICATION_CREDENTIALS and configure-docker if set
+# * run //prow:release-push to build and push prow images
+# * update all the cluster/*.yaml files to use the new image tags
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# TODO(fejta): rewrite this in a better language REAL SOON
```
Comment (BenTheElder) [inline prow/autobump.sh]: 🤔 this usually stays there until it gets removed and the script is 40000 lines /s 😉

File: prow/bump.sh [hunk 2]
```diff
@@ -53,10 +49,19 @@ if ! ($SED --version 2>&1 | grep -q GNU); then
   exit 1
 fi
 
+TAC=tac
+if which gtac &>/dev/null; then
+  TAC=gtac
+fi
+if ! which "$TAC" &>/dev/null; then
+  echo "tac (reverse cat) required. If on OS X then brew install coreutils." >&2
```
Comment (BenTheElder) [inline prow/bump.sh]: ultra-tiny-nitty-knit: `'brew install coreutils'`, it's nice to delineate the command

File: prow/github/client.go [hunk 3]
```diff
@@ -955,6 +955,39 @@ func (c *Client) CreatePullRequest(org, repo, title, body, head, base string, ca
...
+func (c *Client) UpdatePullRequest(org, repo string, number int, title, body *string, open *bool, branch *string, canModify *bool) error {
+	c.log("UpdatePullRequest", org, repo, title)
+	data := struct {
+		State *string `json:"state,omitempty"`
+		Title *string `json:"title,omitempty"`
+		Body  *string `json:"body,omitempty"`
+		Base  *string `json:"base,omitempty"`
+		// MaintainerCanModify allows maintainers of the repo to modify this
+		// pull request, eg. push changes to it before merging.
+		MaintainerCanModify *bool `json:"maintainer_can_modify,omitempty"`
```
Comment (BenTheElder) [inline prow/github/client.go]: 👍 FWIW, this option works pretty well.

File: prow/push.sh [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+color-error() { # Light red
+  echo -e "\x1B[91m${@}\x1B[0m"
+}
+
+color-target() { # Bold cyan
+  echo -e "\x1B[1;33m${@}\x1B[0m"
+}
+
+# darwin is great
```
Comment (BenTheElder) [inline prow/push.sh]: it really is

File: robots/pr-creator/main.go [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+	return o
+}
+
+func main() {
+	o := optionsFromFlags()
+	if err := o.validate(); err != nil {
+		logrus.WithError(err).Fatal("bad flags")
+	}
+
+	var jamesBond config.SecretAgent
```
Comment (BenTheElder) [inline robots/pr-creator/main.go]: love it 🕵️

File: prow/autobump.sh (hunk 1 above)
Comment (BenTheElder) [reply prow/autobump.sh]: lol 😂 
this is fine :-)

**Comment by fejta** [comment]
that was a weird flake
/meowie boxes

**Comment by fejta** [comment]
Ben, add the stamp and/or ptal?

**Pull request merged**